examples/motoko_examples/quicksort/canisters/motoko/Quicksort.mo

```diff
@@ -0,0 +1,55 @@
+import Array "mo:base/Array";
+import Order "mo:base/Order";
+import Int "mo:base/Int";
+
+module Quicksort {
+
+  type Order = Order.Order;
+
+  // Sort the elements of an array using the given comparison function.
+  public func sortBy<X>(xs : [X], f : (X, X) -> Order) : [X] {
```

Comment:
The Motoko and Rust algorithms are not identical, there are various differences

Reply:
I've done my best to align the two algorithms. There's one small difference with how I do the swap because in rust you need two mutable references to the same array. Doing that myself would require an unsafe block so I just used the builtin swap method. I think it's essentially equivalent under the hood, it just doesn't look that way at first glance.

TLDR; I think they're as close as they can get now. I'll let you evaluate that though and mark this thread resolved if you feel confident.